Local/Msync.py

```diff
@@ -14,41 +17,123 @@ def listening_for(self, message):
     def handle(self,message):
         try:
             target = message['metadata']['source_channel']
-            opts = {'target':target}
+            self.opts = {'target':target}
+            self.message = message
         except IndexError:
             logger.error('Could not identify message source in message: %s' \
                          % str(message))
-        arg = self.parse_args(message)
 
-        if arg == None:
-            response = self.get_current_msync()
-            self.reply(message, response, opts)
+        # A python workaround for case statements
+        dispatcher = {'getver':self._get_single_version,
+                     'olderthan':self._compare_to_version,
+                     'newerthan':self._compare_to_version,
+                     'current':self._compare_to_version,
+                     'version':self._get_current_msync}
 
-        elif arg == "getver":
-            modname = None
-            try:
-                modname = message['text'].split()[2]
-            except IndexError:
-                self.reply(message, "Could not find a module name in your request.", opts)
-
-            if modname == None:
-                self.reply(message,self.get_help(),opts)
-            else:
-                response = self.get_single_version(modname)
-                self.reply(message,response,opts)
-        return
+        args = message['text'].split()
+        if len(args) > 1:
+            command = args[1]
+        else:
+            command = 'version'
+        logger.debug('%s requested. Dispatching %s with args: %s' % (command,
+                                                                     str(dispatcher[command]),
+                                                                     str(args)))
+        response = dispatcher[command](*args)
+        # Returning things other than a str to chat really freaks legobot out.
+        # Someone should fix that.
+        self.reply(message,str(response),self.opts)
 
     def get_name(self):
         return 'msync'
 
     def get_help(self):
-        return 'Discover information about the status of modulesync on managed\
- repositories. Usage: !msync [getver modulename].'
+        return 'Discover information about the status of modulesync on managed '\
+                'repositories. Usage: !msync. Fore more info, see the docs at '\
+                'https://github.com/voxpupuli/thevoxfox/tree/master/docs/msync.md'
```

Comment:
We discussed help on IRC , heh you went hog wild. Perhaps we should still have a _minimal_ output of args, with the docs expanding that.

Great work and LGTM